services/crawlingService.js

```diff
@@ -1,54 +1,361 @@
 const webdriver = require('selenium-webdriver');
 const chrome = require('selenium-webdriver/chrome');
-const { By, Key } = require('selenium-webdriver');
+const { By, until } = require('selenium-webdriver');
+const axios = require('axios');
+const xml2js = require('xml2js');
+const Building = require('../model/Building');
 
-async function run() {
-  const service = new chrome.ServiceBuilder('/Users/khan/Desktop/Auxious/Auxious-Server/services/chromedriver').build();
+async function runAuctionCrawling() {
+  const service = new chrome.ServiceBuilder(
+    '/Users/khan/Desktop/Auxious/Auxious-Server/services/chromedriver',
+  ).build();
   chrome.setDefaultService(service);
-  const driver = await new webdriver.Builder()
-    .forBrowser('chrome')
-    .build();
+  const driver = await new webdriver.Builder().forBrowser('chrome').build();
 
   await driver.manage().setTimeouts({
     implicit: 10000,
     pageLoad: 30000,
     script: 30000,
   });
 
+  await connectSetAuctionSearchOptions(driver);
+  await loginAuctionPage(driver);
+  await autoSearchNextPage(driver);
+  await driver.close();
+}
 
-  await driver.get('http://www.my-auction.co.kr/auction/search.php');
-  const selectRegionMenu = await driver.findElement(By.id('address1_01'));
-  await selectRegionMenu.sendKeys('서울특별시');
-  const selectForSaleTypeCheckBox = await driver.findElement(By.className('t_color01')).findElement(By.tagName('input'));
-  await selectForSaleTypeCheckBox.click();
-  const totaSearchButton = await driver.findElement(By.id('search_btn')).findElement(By.tagName('button'));
-  await totaSearchButton.click();
+async function autoSearchNextPage(driver) {
+  const firstPage = await driver.findElement(By.className('ytb_1st'));
+  const firstPageUrl = await firstPage.getAttribute('href');
+  await driver.get(firstPageUrl);
+
+  const orderButtonList = await driver
+    .findElement(By.id('sort_btn'))
+    .findElement(By.tagName('ul'))
+    .findElements(By.tagName('li'));
+  const dateOrderButton = await orderButtonList[4].findElements(
+    By.tagName('a'),
+  );
+
+  await dateOrderButton[1].click();
 
-  const loginPageButton = await driver.findElement(By.className('logon')).findElement(By.tagName('a'));
+  let newUrl = firstPageUrl;
+  newUrl = newUrl.replace('page=1', 'page=0');
+
+  for (let i = 1; i < 31; i++) {
+    newUrl = newUrl.replace(`page=${i}`, `page=${i + 1}`);
+    await driver.get(newUrl);
+    await connectToAuctionDetailPage(driver);
+  }
+}
+
+async function connectToAuctionDetailPage(driver) {
+  const auctionListTable = await driver.findElement(
+    By.className('tbl_auction_list'),
+  );
+  const auctionList = await auctionListTable.findElements(By.tagName('tr'));
```

Comment:
```suggestion
  const auctionListTable = await driver.findElement(
    By.className('tbl_auction_list'),
  );
  
  const auctionList = await auctionListTable.findElements(By.tagName('tr'));
```

두 줄 넘어가는 코드의 경우 다음 코드를 쓸때 공백 한칸 주시는게 읽기 좋은거 같습니다.

Reply:
스타일로 하기로 협의를 해야될거같은데 그거는 세현님 의견을 물어봐야될것 같네요 저는 동의합니다.

Reply:
개인적으로는 두 줄 이상 코드를 사용할 때는 위 아래로 한칸씩 공백을 주는 것을 선호하는데 세현님도 동의하시면 스타일 추가사항으로 노션에 반영하겠습니다

Reply:
저도 두 줄 이상 코드를 사용할 때는 위 아래로 한칸씩 공백을 주는 것 좋은 것 같습니다! 동의합니다.